c++/src/kj/compat/http.c++

```diff
@@ -1110,6 +1112,150 @@ kj::String HttpHeaders::toString() const {
   return serialize(nullptr, nullptr, nullptr, nullptr);
 }
 
+// -----------------------------------------------------------------------------
+
+
+namespace {
+
+// RFC9110 section 14.1: https://www.rfc-editor.org/rfc/rfc9110#section-14.1
```

Comment:
It'd help to add a few comments here summarizing what we're parsing so that people don't have to open the spec. Like, what does the overall text typically look like, and what's the difference between an int range and a suffix range? You don't have to go into full detail, people can refer to the spec for that.